src/client/js/error.js

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const genericErrorMessage = 'Something went wrong. Please try again or come back later.'
+const ErrorActionTypes = {
+  None: 'none',
+  Toast: 'toast'
+}
+const defaultConfig = { action: ErrorActionTypes.None }
+
+class ClientError extends Error {
+  /**
+   * @param {string} message
+   * @param {Array<{ action: string, targetHref: string }>} config
+   */
+  constructor (message, config = defaultConfig) {
+    super(message, config)
+
+    this.config = config
+    this.message = message || genericErrorMessage
```

Comment:
q: why fall back to `genericErrorMessage` here, versus as a constructor default (like `defaultConfig`)?
Probably doesn't matter, just personally curious I guess. And whether it makes a difference since we call `super(message, config)` with a potentially falsy `message`.

Reply:
The thinking was: Passing `null` the arg would not fall back to a default value. Only no value or `undefined` would make that happen and personally, I’d like the option of being able to pass `null` like now [here](https://github.com/mozilla/blurts-server/pull/3010/files#diff-135402cfb34bfc8e8f3df69241e60d14eec552b4edfdbbf87996209784f60c4aR60). Adding `message` to super is definitely redundant in this case, though. Addressed this in commit https://github.com/mozilla/blurts-server/pull/3010/commits/e62334fb364d7e4de8386c6838fcfb43d7896d66 — WDYT?